lib/common/zstd_internal.h

```diff
@@ -56,6 +56,48 @@ extern "C" {
 #define CHECK_F(f) { size_t const errcod = f; if (ERR_isError(errcod)) return errcod; }  /* check and Forward error code */
 #define CHECK_E(f, e) { size_t const errcod = f; if (ERR_isError(errcod)) return ERROR(e); }  /* check and send Error code */
 
+/**
+ * Return the specified error if the condition evaluates to true.
+ *
+ * In debug modes, prints additional information.
+ */
+#define RETURN_ERROR_IF(cond, err, ...) \
```

Comment:
So, the main reason why `RETURN_ERROR_IF(...)` is not a simple `if (cond) RETURN_ERROR(...)`
is to be able to log `cond` itself ?

That would a good enough reason, since I don't see any other way around.

Maybe add a comment about that in the macro description.

Reply:
Yep, that's right. Happy to add a comment.